src/server/middleware/metric/metric.go

```diff
@@ -1,25 +1,65 @@
 package metric
 
 import (
+	"context"
 	"net/http"
 	"strconv"
+	"strings"
 	"time"
 
 	"github.com/goharbor/harbor/src/core/config"
 	"github.com/goharbor/harbor/src/lib"
 	"github.com/goharbor/harbor/src/lib/metric"
 )
 
+// ContextOpIDKey ...
+type contextOpIDKey struct{}
+
+const (
+	// CatalogOperationID ...
+	CatalogOperationID = "v2_catalog"
+	// ListTagOperationID ...
+	ListTagOperationID = "v2_tags"
+	// ManifestOperationID ...
+	ManifestOperationID = "v2_manifest"
+	// BlobsOperationID ...
+	BlobsOperationID = "v2_blob"
+	// BlobsUploadOperationID ...
+	BlobsUploadOperationID = "v2_blob_upload"
+	// OthersOperationID ...
+	OthersOperationID = "v2_others"
+)
+
+// SetMetricOpID used to set operation ID for metrics
+func SetMetricOpID(ctx context.Context, value string) {
+	if config.Metric().Enabled {
+		v := ctx.Value(contextOpIDKey{}).(*string)
+		*v = value
+	}
+}
+
+func isChartMuseumURL(url string) bool {
+	return strings.HasPrefix(url, "/chartrepo/") || strings.HasPrefix(url, "/api/chartrepo/")
+}
+
 func instrumentHandler(next http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		now, url := time.Now(), r.URL.EscapedPath()
-
-		metric.TotalInFlightGauge.WithLabelValues(url).Inc()
-		defer metric.TotalInFlightGauge.WithLabelValues(url).Dec()
-		rc := lib.NewResponseRecorder(w)
-		next.ServeHTTP(rc, r)
-		metric.TotalReqDurSummary.WithLabelValues(r.Method, url).Observe(time.Since(now).Seconds())
-		metric.TotalReqCnt.WithLabelValues(r.Method, strconv.Itoa(rc.StatusCode), url).Inc()
+		metric.TotalInFlightGauge.Inc()
+		defer metric.TotalInFlightGauge.Dec()
+		now, rc, src := time.Now(), lib.NewResponseRecorder(w), ""
+		if isChartMuseumURL(r.URL.Path) {
+			src = "chartmuseum"
+		}
```

Comment:
Please only do this checking when the `src` (may it named `operation`) is an empty string after the `next.ServeHTTP` called.